lib/util.js

```diff
@@ -557,9 +565,14 @@ function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
 
 
 function formatTypedArray(ctx, value, recurseTimes, visibleKeys, keys) {
-  var output = new Array(value.length);
-  for (var i = 0, l = value.length; i < l; ++i)
+  const maxLength = Math.min(Math.max(0, ctx.maxArrayLength), value.length);
+  const remaining = value.length - maxLength;
+  var output = new Array(maxLength);
```

Comment:
Kept as `var` to remain consistent with every other instance of `var output =` in this file. Would rather maintain that consistency and fix those up all at once separately.